photon-api/src/main/scala/com/linkedin/photon/ml/optimization/RandomEffectOptimizationTracker.scala

```diff
@@ -89,57 +97,56 @@ protected[ml] class RandomEffectOptimizationTracker(val optimizationStatesTracke
    *
    * @return A map of convergence reason to its count of occurrence from the random effect optimization problems
    */
-  def countConvergenceReasons: Map[String, Int] = {
-    optimizationStatesTrackers.map { optimizationStatesTracker =>
-      if (optimizationStatesTracker.convergenceReason.isDefined) {
-        (optimizationStatesTracker.convergenceReason.get.reason, 1)
-      } else {
-        (RandomEffectOptimizationTracker.NOT_CONVERGED, 1)
-      }
-    }.reduceByKey(_ + _).collectAsMap()
-  }
+  def countConvergenceReasons: Map[String, Int] = optimizationStatesTrackers.map { optimizationStatesTracker =>
+      (optimizationStatesTracker
+        .convergenceReason
+        .map(_.reason)
+        .getOrElse(RandomEffectOptimizationTracker.NOT_CONVERGED), 1)
```

Comment:
What are other possible statuses? Should we count e.g. only the successes instead, and warn in case of any status but success?